### ISSUE TITLE
Accept input from URL

### ISSUE THREAD

**Issue by n4bb12** [sharkdp/insect#161]
Similar to `https://www.google.de/search?q=insect`

it would be handy if input could be passed in the URL.
This would allow me to start calculations from the command line, bookmark results and share them with friends and coworkers.

Awesome app 🥇

**Comment by sharkdp** [comment]
Thank you for the feedback!

This is already supported (via OpenSearch):

https://insect.sh/?q=cos(pi)

You should be also be able to use this by typing "insect.sh" in your browsers address bar and pressing <TAB> to open a search bar.

**Comment by n4bb12** [comment]
Exactly what I was looking for!

**Comment by n4bb12** [comment]
Hmm, would it be possible to keep the URL updated with the input, too?

**Comment by sharkdp** [comment]
It's certainly possible somehow. Is this a method that is commonly used? To change the URL in single-page web applications? Does this work without triggering some kind of reload-event in the browser?

**Comment by n4bb12** [comment]
Typically it's used in all types of search apps.

[GitHub Issues](https://github.com/Microsoft/vscode/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) writes your filters into the URL, too.

In an SPA context it's possible with `window.history`. You can `pushState` or `replaceState`.
https://developer.mozilla.org/en-US/docs/Web/API/History_API#Adding_and_modifying_history_entries

You can see it in action here:
https://items.projectceleste.com/

**Comment by sharkdp** [comment]
Ok, sounds good! It should be relatively straightforward to add this in `web/index.html` in the `interpret` function.

**Comment by n4bb12** [comment]
Want me do create a PR?

**Comment by sharkdp** [comment]
That would be great! Contributions are always welcome.

**Comment by n4bb12** [comment]
The implementation in the PR really is very naive as it does not account for context variables.

For example when typing the following,
```
r = 6000km
vol = 4/3 * pi * r^3
density = 5 g/cm^3
vol * density -> kg
```

the URL ends up being

```
http://127.0.0.1:8080/?q=vol%20*%20density%20-%3E%20kg
```

But when hitting refresh, the output is:

```
Unknown identifier: vol
```

I feel like I don't have enough context knowledge in order to make this work, too. I'd kinda have to find out what variables the latest input depends on and add those to the URL, too.

**Comment by sharkdp** [comment]
> The implementation in the PR really is very naive as it does not account for context variables.

To fully support this, we would have to serialize/deserialize the global `insectEnv` variable and add it to the URL. This object includes quite a bit of information (like abstract syntax trees for user-defined functions), so the URL will probably get really long.

I think I'd be okay with the current implementation. It is still quite valuable without the full context.

**Comment by n4bb12** [comment]
Just a small detail, but I just noticed that `clear` and `quit` are treated separately and don't reset the URL.
Do you think it would be useful to add this? I already know what to do.

**Comment by sharkdp** [comment]
Good point. Yes, it would be great if `clear` and `quit` would also reset the URL.
